### PR TITLE
Testing/Module-module_Tests

### LINKED ISSUE #145: Testing Module Module
## The story:
Adding unit tests to the Module Module.

---

## What we should do:
To get functional and branch percentages up, we need to make sure all of the resolver methods that can be tested are tested and as much branching (edge cases) covered as much as possible.

---

## To be successful we need to:
 - [x] Import the module in the utils/TestClient.ts file
 - [x]  Create and run a test for each method of the service file in the tests/entities folder
 - [x] Test every possible wrong scenario
 - [x] Ensure that each methods fullfill their API contract

### PR DESCRIPTION
**Pull request by LeeYang2019**
- Fixes #145 

## Changes
- Change 1: cleaned up and re-organized code in Module.test.ts file

## PR Checklist
- [x] Does your PR contain relevant tests that pass?
- [x] Is your code linted? `yarn lint` or `yarn lint:fix`
- [x] Does your code build correctly in ALL projects? `yarn build`

### CODE REVIEW

**Comment by LeeYang2019** [comment]
Was working with a corrupt branch, so closed out the other PR and opened a new one here with a new branch from develop. Please review and tell me how I can improve my tests. Thanks.

**Comment by LeeYang2019** [comment]
Hey @quimgv, can you re-review this when you have a chance?

File: packages/api/src/UserModule/index.ts [hunk 1]
```diff
@@ -0,0 +1,9 @@
+import { Module } from '@nestjs/common';
+import { TypeOrmModule } from '@nestjs/typeorm';
+
+import { UserModule } from './UserModule.entity';
+
+@Module({
+  imports: [TypeOrmModule.forFeature([UserModule])]
+})
+export class UserModuleModule {}
```
Comment (ghost) [inline packages/api/src/UserModule/index.ts]: That's an unfortunate class name with "Module" occurring twice, 
it's OK for now but we should avoid using "Module" in entity names.

Comment (quimgv) [reply packages/api/src/UserModule/index.ts]: How can we name it?

Comment (quimgv) [reply packages/api/src/UserModule/index.ts]: We also have **ModuleModule**

File: packages/api/tests/entities/Module.test.ts [hunk 2]
```diff
@@ -83,31 +104,67 @@ describe('Module entity', () => {
...
+
+    it('should not allow to join a module twice', async () => {
+      expect.assertions(1);
+
+      const module = await TestClient.createModule({
+        ...moduleInput,
+        pathId: path.id
+      });
+
+      await TestClient.createUser();
```
Comment (quimgv) [inline packages/api/tests/entities/Module.test.ts]: What is this for?

File: packages/api/tests/entities/Module.test.ts [hunk 3]
```diff
@@ -83,31 +104,67 @@ describe('Module entity', () => {
...
+    });
+  });
 
+  describe('Mutation: Join Module', () => {
+    beforeEach(async () => {
+      await setup();
+      path = await TestClient.createPath(pathInput);
+    });
+
+    it('join a module successfully', async () => {
```
Comment (quimgv) [inline packages/api/tests/entities/Module.test.ts]: add "should" to keep it consistent with the others please

File: packages/api/tests/entities/Module.test.ts [hunk 4]
```diff
@@ -67,14 +74,28 @@ describe('Module entity', () => {
...
     it('should update a module successfully', async () => {
-      expect.assertions(6);
-      const modulePrev = await TestClient.createModule({ ...moduleInput, name: 'previous', pathId: path.id });
-      const module = await TestClient.createModule({ ...moduleInput, pathId: path.id });
-      const path2 = await TestClient.createPath({ ...pathInput, name: 'path 2' }, {name: 'name 2', displayName: 'display name 2'});
+      expect.assertions(2);
+
+      const modulePrev = await TestClient.createModule({
+        ...moduleInput,
+        name: 'previous',
```
Comment (quimgv) [inline packages/api/tests/entities/Module.test.ts:85]: I think you are still using prettier and makes de reviews harder to review having to pay attention to changes that aren't actually changes.

**Comment by quimgv** [comment]
Good work @LeeYang2019 ! I added a few small comments :) 

@shaneSheridan let us know if you come up with a better name for that and we can get it done here.

@LeeYang2019 please look into the prettier issue, you can configure it in settings to not being detected by this project but detected by others.

File: packages/api/src/UserModule/index.ts (hunk 1 above)
Comment (ghost) [reply packages/api/src/UserModule/index.ts]: Just an informational comment for now; I don't have a suggestion at this time.

**Comment by tristanMatthias** [comment]
What's the go with this one @LeeYang2019 , @quimgv ? Been here for a while, would like to get it on the move if possible.

**Comment by quimgv** [comment]
@tristanMatthias  I reviewed it and waiting for @LeeYang2019 's changes or comments.

**Comment by tristanMatthias** [comment]
Ok. @LeeYang2019 I know you've mentioned you've been doing other things. Can you please either fix these changes or communicate we need to put someone else on it.

File: packages/api/tests/entities/Module.test.ts (hunk 2 above)
Comment (LeeYang2019) [reply packages/api/tests/entities/Module.test.ts]: Sorry. Thought I had to create a user before I can test the joinModule function. This has been removed. Thanks @quimgv .

File: packages/api/tests/entities/Module.test.ts (hunk 3 above)
Comment (LeeYang2019) [reply packages/api/tests/entities/Module.test.ts]: This was an oversight on my part. This has been updated. Thanks.

File: packages/api/tests/entities/Module.test.ts (hunk 4 above)
Comment (LeeYang2019) [reply packages/api/tests/entities/Module.test.ts:85]: I thought the vscode settings you gave me would have helped with this issue. Since then, whenever I run yarn lint I see no issues. I've since uninstalled prettier. Not sure how that will help. I will look for a way to toggle it on and off when working on this project.

**Comment by LeeYang2019** [comment]
Hey @shaneSheridan and/or @quimgv, I apologize for the inconvenience. I should have had this done a few days back. Can either of you review and let me know if any issue is still persisting? 

CC: @tristanMatthias

**Comment by LeeYang2019** [comment]
Nevermind. I see a few conflicts that must be resolved. Will work on resolving them.

**Comment by LeeYang2019** [comment]
So, I see there are two conflicts that must be resolved. As these conflicts are not related to the testing I completed on the Module.test.ts file, how should I resolve them? @shaneSheridan, @tristanMatthias, @quimgv.

**Comment by quimgv** [comment]
Yes please, solve them and let me know if you have any doubt. Thanks!

**Comment by LeeYang2019** [comment]
Hey @quimgv, I've resolved the conflicts. Please review and let me know what else needs to be done. Sorry, I attempted to merge this. Saw a bright green button and could not resist.
